### PR TITLE
feat: add support for joining as a group

### PR DESCRIPTION
**Pull request by 0c370t**

### CODE REVIEW

**Comment by 0c370t** [comment]
Task linked: [CU-2fzvyy4 Add support for joining a scrim as a group](https://app.clickup.com/t/2fzvyy4)

**Comment by 0c370t** [comment]
image
image
image

File: clients/web/src/lib/components/organisms/scrims/modals/JoinScrimModal.svelte [hunk 1]
```diff
@@ -1,42 +1,74 @@
...
+        joining = true;
         await joinScrimMutation({
             scrimId: scrimId,
         });
         visible = false;
     }
 
-    function joinAsNewGroup() {
+    async function joinAsNewGroup() {
+        joiningAsGroup = false;
```
Comment (gankoji) [inline clients/web/src/lib/components/organisms/scrims/modals/JoinScrimModal.svelte:23]:
```suggestion
        joiningWithExistingGroup = false;
```
same, trying to make life easy by suggesting it at all points of use

File: clients/web/src/lib/components/organisms/scrims/modals/JoinScrimModal.svelte [hunk 2]
```diff
@@ -1,42 +1,74 @@
...
+    import {slide} from "svelte/transition";
     import {joinScrimMutation} from "$lib/api";
-    import {Modal} from "$lib/components";
+    import {Modal, toasts} from "$lib/components";
 
     export let visible = true;
     export let scrimId: string;
 
+    let groupCode: string;
+    let joiningAsGroup: boolean = false;
```
Comment (gankoji) [inline clients/web/src/lib/components/organisms/scrims/modals/JoinScrimModal.svelte:10]:
```suggestion
    let joiningWithExistingGroup: boolean = false;
```
naming nitpick

File: clients/web/src/lib/components/organisms/scrims/modals/JoinScrimModal.svelte [hunk 3]
```diff
@@ -1,42 +1,74 @@
...
+    let joining: boolean = false;
 
     async function joinSolo() {
+        joiningAsGroup = false;
```
Comment (gankoji) [inline clients/web/src/lib/components/organisms/scrims/modals/JoinScrimModal.svelte:14]:
```suggestion
        joiningWithExistingGroup = false;
```
same, trying to make life easy by suggesting it at all points of use

**Review by gankoji — APPROVED** [review]

File: clients/web/src/lib/components/organisms/scrims/modals/JoinScrimModal.svelte [hunk 4]
```diff
@@ -1,42 +1,74 @@
...
 
         </div>
         <div class="divider">or</div>
         <div class="flex items-center gap-2">
-            <h3 class="flex-1">Play Together <span class="text-gray-300 text-sm">(Coming Soon)</span></h3>
-            <button on:click={joinAsNewGroup} title="Coming Soon" disabled>Create group</button>
-            <button on:click={joinExistingGroup} title="Coming Soon" disabled>Join group</button>
+            <h3 class="flex-1">Play Together</h3>
+            <button on:click={joinAsNewGroup} disabled={joining}>Create group</button>
+            <button on:click={() => { joiningAsGroup = !joiningAsGroup }} disabled={joining}>Join group</button>
```
Comment (0c370t) [inline clients/web/src/lib/components/organisms/scrims/modals/JoinScrimModal.svelte:64]:
```suggestion
            <button on:click={() => { joinWithExistinGroup = !joinWithExistinGroup }} disabled={joining}>Join group</button>
```

@gankoji to answer your question; here

**Pull request merged**